File: csharp/1636.sort-array-by-increasing-frequency.cs
/*
 * @lc app=leetcode id=1636 lang=csharp
 *
 * [1636] Sort Array by Increasing Frequency
 */

using System.Linq;

// @lc code=start
public class Solution {
    public int[] FrequencySort(int[] nums) {
        return nums
            .GroupBy(x => x, (x, a) => (Value: x, Count: a.Count()))
            .OrderBy(t => t.Count)
            .ThenByDescending(t => t.Value)
            .SelectMany(t => Enumerable.Repeat(t.Value, t.Count))
            .ToArray();
    }
}
// @lc code=end

File: java/1636.sort-array-by-increasing-frequency.java
/*
 * @lc app=leetcode id=1636 lang=java
 *
 * [1636] Sort Array by Increasing Frequency
 */

// @lc code=start

import java.util.Arrays;

class Solution {
    public int[] frequencySort(int[] nums) {
        int[] count = new int[201];
        for (int x : nums) {
            ++count[x + 100];
        }
        return Arrays
            .stream(nums)
            .boxed()
            .sorted(java.util.Comparator.comparingInt((Integer x) -> count[x + 100]).thenComparingInt(x -> -x))
            .mapToInt(x -> x)
            .toArray();
    }
}
// @lc code=end

